<a href="https://colab.research.google.com/github/abhishekhkumarsharma/Assignment_Solutions.ipynb/blob/main/Statistical%20Summaries%20and%20Reporting.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Resources

### Created reports and summaries using statistical techniques.

##  Statistical Summaries and Reporting

Beyond basic descriptive statistics, we can delve into more detailed statistical measures and report on their implications. This section focuses on quantifying characteristics of the data and relationships between variables.

In [17]:
import numpy as np

print("### Advanced Descriptive Statistics Report ###")
print("------------------------------------------\n")

# Calculate skewness and kurtosis
skewness_col1 = df['col1'].skew()
kurtosis_col1 = df['col1'].kurt()
skewness_col2 = df['col2'].skew()
kurtosis_col2 = df['col2'].kurt()

print(f"* **Column 'col1'**:")
print(f"  - Mean: {df['col1'].mean():.2f}")
print(f"  - Median: {df['col1'].median():.2f}")
print(f"  - Std Dev: {df['col1'].std():.2f}")
print(f"  - Skewness: {skewness_col1:.2f} (Measures asymmetry of the distribution. 0 indicates perfect symmetry.)")
print(f"  - Kurtosis: {kurtosis_col1:.2f} (Measures 'tailedness' of the distribution. >0 indicates heavy tails/outliers.)")

print(f"\n* **Column 'col2'**:")
print(f"  - Mean: {df['col2'].mean():.2f}")
print(f"  - Median: {df['col2'].median():.2f}")
print(f"  - Std Dev: {df['col2'].std():.2f}")
print(f"  - Skewness: {skewness_col2:.2f}")
print(f"  - Kurtosis: {kurtosis_col2:.2f}")

print("\n--- Guidance for Interpretation ---")
print("**Skewness**:")
print("- If skewness is negative, the data are 'skewed to the left' (tail is longer on the left side). The mean is typically less than the median.")
print("- If skewness is positive, the data are 'skewed to the right' (tail is longer on the right side). The mean is typically greater than the median.")
print("**Kurtosis**:")
print("- High kurtosis (leptokurtic) indicates heavy tails or outliers.")
print("- Low kurtosis (platykurtic) indicates light tails or lack of outliers.")
print("- Mesokurtic (kurtosis near 0 for excess kurtosis, or 3 for Pearson's kurtosis) is similar to a normal distribution.")

### Advanced Descriptive Statistics Report ###
------------------------------------------

* **Column 'col1'**:
  - Mean: 48.00
  - Median: 42.50
  - Std Dev: 33.57
  - Skewness: 0.23 (Measures asymmetry of the distribution. 0 indicates perfect symmetry.)
  - Kurtosis: -1.72 (Measures 'tailedness' of the distribution. >0 indicates heavy tails/outliers.)

* **Column 'col2'**:
  - Mean: 0.44
  - Median: 0.41
  - Std Dev: 0.30
  - Skewness: 0.17
  - Kurtosis: -1.55

--- Guidance for Interpretation ---
**Skewness**:
- If skewness is negative, the data are 'skewed to the left' (tail is longer on the left side). The mean is typically less than the median.
- If skewness is positive, the data are 'skewed to the right' (tail is longer on the right side). The mean is typically greater than the median.
**Kurtosis**:
- High kurtosis (leptokurtic) indicates heavy tails or outliers.
- Low kurtosis (platykurtic) indicates light tails or lack of outliers.
- Mesokurtic (kurtosis near 0 for excess kurto

In [19]:
print("\n### Correlation Analysis Report ###")
print("-----------------------------------\n")

# Calculate the Pearson correlation coefficient between col1 and col2
correlation = df['col1'].corr(df['col2'])

print(f"* **Pearson Correlation between 'col1' and 'col2'**: {correlation:.2f}\n")

print("--- Guidance for Interpretation ---")
print("**Correlation Coefficient (r)**:")
print("- Ranges from -1 to +1.")
print("- **+1**: Perfect positive linear relationship (as one variable increases, the other increases proportionally).")
print("- **-1**: Perfect negative linear relationship (as one variable increases, the other decreases proportionally).")
print("- **0**: No linear relationship.")
print("- Generally, |r| values between 0.1 and 0.3 are considered weak, 0.3 and 0.5 moderate, and > 0.5 strong.")
print("\n*Important Note: Correlation does not imply causation!*")


### Correlation Analysis Report ###
-----------------------------------

* **Pearson Correlation between 'col1' and 'col2'**: -0.07

--- Guidance for Interpretation ---
**Correlation Coefficient (r)**:
- Ranges from -1 to +1.
- **+1**: Perfect positive linear relationship (as one variable increases, the other increases proportionally).
- **-1**: Perfect negative linear relationship (as one variable increases, the other decreases proportionally).
- **0**: No linear relationship.
- Generally, |r| values between 0.1 and 0.3 are considered weak, 0.3 and 0.5 moderate, and > 0.5 strong.

*Important Note: Correlation does not imply causation!*


## Summary of Statistical Findings

This section consolidates the statistical analyses into a concise report, highlighting key metrics and their interpretations.

### Report for Dataset Analysis:

**1. Advanced Descriptive Statistics:**

*   **`col1`**:
    *   Mean: `48.00`
    *   Median: `42.50`
    *   Standard Deviation: `33.57`
    *   Skewness: `0.23`. This indicates `a slight positive skew, suggesting a slightly longer tail of higher values`.
    *   Kurtosis: `-1.72`. This suggests `lighter tails than a normal distribution (platykurtic), possibly indicating fewer extreme values or outliers compared to a normal distribution`.

*   **`col2`**:
    *   Mean: `0.44`
    *   Median: `0.41`
    *   Standard Deviation: `0.30`
    *   Skewness: `0.17`. This indicates `a slight positive skew`.
    *   Kurtosis: `-1.55`. This suggests `lighter tails than a normal distribution (platykurtic)`.

**2. Relationship Analysis (Correlation):**

*   The Pearson correlation coefficient between `col1` and `col2` is `-0.07`. This indicates `a very weak to almost no linear relationship between the two variables. As col1 changes, col2 does not show a clear linear tendency to increase or decrease`.

**Implications for Decision-Making:**

Based on these statistical insights, decision-makers can understand the underlying characteristics of each variable and the nature of their linear relationship. For instance, if `col1` and `col2` represented critical business metrics, this report would inform whether they move in tandem, opposition, or independently, which is vital for strategic planning and resource allocation. For example, a strong positive correlation could suggest interdependent processes, while a weak correlation might imply that factors influencing one variable do not significantly affect the other.